doc/user/content/transform-data/patterns/window-functions.md

```diff
@@ -1,133 +0,0 @@
----
-title: "Window functions (OVER clause)"
-description: "How to efficiently handle queries that require window functions in Materialize."
-aliases:
-  - /sql/patterns/window-functions/
-menu:
-  main:
-    parent: 'sql-patterns'
----
-
-Materialize offers a wide range of [window functions](/sql/functions/#window-functions), such as `LAG`, `LEAD`, `ROW_NUMBER`, and `FIRST_VALUE`, to support a variety of complex query patterns. This page discusses certain window function query patterns where rewriting your query to not use window functions can yield a better performance.
-
-Currently, Materialize implements window functions by recomputing results for an entire window partition when an input record is added, removed or changed in the partition. This means that when a new batch of input data arrives (that is, every second) the system has to perform an amount of computation proportional to the total size of the touched partitions (where the partitions are determined by the `PARTITION BY` clause inside the `OVER` clause of your window function call). For example, if in a certain second 20 input records change, and these records belong to 10 different partitions, and the average sizes of these partitions is 100, then the system will perform work proportional to computing window function results for 10\*100=1000 rows.
-
-As a rule of thumb, if the total size of all touched window partitions is at most 1000000 rows per second, then the system should be able to keep up with the input data as it arrives. However, if you find that your use case has higher performance requirements, then consider rewriting your query to not use window functions. This page helps with such rewrites.
```

Comment:
👍 will single-source an add it in multiple places.

Reply:
It's looking great, thank you very much!

One more minor thing came to my mind. Since it's often not actually possible to rewrite a window function query to not use window functions (because not all window function usage fits these patterns that are explained in these docs), we could add an extra sentence after "consider rewriting your query to not use window functions" that says something like "If it's not possible to rewrite your query to not use window functions, and you find the performance of window functions insufficient for your use case, then please contact us.". This would serve two purposes:
1. It would increase the chances that we'd get to know about these situations rather than people just silently giving up. This information is good input for determining the priority of further window function work.
2. There are some more tricks up our sleeves for rewriting window function queries to increase their performance. These tricks are a bit unwieldy, so I wouldn't explain them in the docs for now, but we can explain it directly to users on a case-by-case basis. (And then maybe later we'll also add them to the docs if we get a bit more experience with them.)